numpy/_core/src/umath/loops_comparison.dispatch.cpp

```diff
@@ -0,0 +1,638 @@
+#include "numpy/npy_common.h"
+#include "common.hpp"
+#include "simd/simd.h"
+#include "loops_utils.h"
+#include "loops.h"
+#include <hwy/highway.h>
+
+namespace {
+
+namespace hn = hwy::HWY_NAMESPACE;
+
+const hn::ScalableTag<uint8_t>  u8;
+const hn::ScalableTag<int8_t>   s8;
+const hn::ScalableTag<uint16_t> u16;
+const hn::ScalableTag<int16_t>  s16;
+const hn::ScalableTag<uint32_t> u32;
+const hn::ScalableTag<int32_t>  s32;
+const hn::ScalableTag<uint64_t> u64;
+const hn::ScalableTag<int64_t>  s64;
+const hn::ScalableTag<float>    f32;
+const hn::ScalableTag<double>   f64;
+
+using vec_u8  = hn::Vec<decltype(u8)>;
+using vec_s8  = hn::Vec<decltype(s8)>;
+using vec_u16 = hn::Vec<decltype(u16)>;
+using vec_s16 = hn::Vec<decltype(s16)>;
+using vec_u32 = hn::Vec<decltype(u32)>;
+using vec_s32 = hn::Vec<decltype(s32)>;
+using vec_u64 = hn::Vec<decltype(u64)>;
+using vec_s64 = hn::Vec<decltype(s64)>;
+using vec_f32 = hn::Vec<decltype(f32)>;
+using vec_f64 = hn::Vec<decltype(f64)>;
+
+template<typename T>
+struct TagSelector;
+
+template<> struct TagSelector<uint8_t>  { static const auto& value() { return u8;  } };
+template<> struct TagSelector<int8_t>   { static const auto& value() { return s8;  } };
+template<> struct TagSelector<uint16_t> { static const auto& value() { return u16; } };
+template<> struct TagSelector<int16_t>  { static const auto& value() { return s16; } };
+template<> struct TagSelector<uint32_t> { static const auto& value() { return u32; } };
+template<> struct TagSelector<int32_t>  { static const auto& value() { return s32; } };
+template<> struct TagSelector<uint64_t> { static const auto& value() { return u64; } };
+template<> struct TagSelector<int64_t>  { static const auto& value() { return s64; } };
+template<> struct TagSelector<float>    { static const auto& value() { return f32; } };
+template<> struct TagSelector<double>   { static const auto& value() { return f64; } };
+
+template<typename T>
+constexpr const auto& GetTag() {
+    return TagSelector<T>::value();
+}
+
+template <typename T>
+constexpr bool kSupportLane = false;
+
+template <> constexpr bool kSupportLane<uint8_t>  = true;
+template <> constexpr bool kSupportLane<int8_t>   = true;
+template <> constexpr bool kSupportLane<uint16_t> = true;
+template <> constexpr bool kSupportLane<int16_t>  = true;
+template <> constexpr bool kSupportLane<uint32_t> = true;
+template <> constexpr bool kSupportLane<int32_t>  = true;
+template <> constexpr bool kSupportLane<uint64_t> = true;
+template <> constexpr bool kSupportLane<int64_t>  = true;
+template <> constexpr bool kSupportLane<float>    = true;
+template <> constexpr bool kSupportLane<double>   = true;
+
+template <typename T>
+struct OpEq {
+#if NPY_SIMD
+    template <typename V, typename = std::enable_if_t<kSupportLane<T>>>
+    HWY_INLINE HWY_ATTR auto operator()(const V &v)
```

Comment:
```suggestion
    HWY_INLINE auto operator()(const V &v)
```
No need for it I suppose, and even if we decided to move Highway  `start/end` macros can be used instead.